### PR TITLE
Jenkins E2E build should not fail if symlink to junit file already exists

### LINKED ISSUE #9783: Jenkins reports E2E symlink creation failure as build failure
Jenkins reports builds as failed if it fails to create the `junit*.xml to ${WORKSPACE}` symlink in E2E script.

```
+ for junit in '${ARTIFACTS}/junit*.xml'
+ ln -s /var/lib/jenkins/jobs/kubernetes-soak-weekly-deploy-gke/workspace/_artifacts/junit_01.xml /var/lib/jenkins/jobs/kubernetes-soak-weekly-deploy-gke/workspace
ln: failed to create symbolic link `/var/lib/jenkins/jobs/kubernetes-soak-weekly-deploy-gke/workspace/junit_01.xml': File exists
```

### PR DESCRIPTION
**Pull request by saad-ali**
Closes #9783

CC @zmerlynn

### CODE REVIEW

File: hack/jenkins/e2e.sh [hunk 1]
```diff
@@ -227,8 +227,9 @@ fi
 # TODO(zml): We have a bunch of legacy Jenkins configs that are
 # expecting junit*.xml to be in ${WORKSPACE} root and it's Friday
 # afternoon, so just put the junit report where it's expected.
+# If link already exists, non-zero return code should not cause build to fail.
 for junit in ${ARTIFACTS}/junit*.xml; do
-  ln -s ${junit} ${WORKSPACE}
+  ln -s ${junit} ${WORKSPACE} || true
```
Comment (zmerlynn) [inline hack/jenkins/e2e.sh]: You could just use "ln -s -f" .. that way you ensure it's the link you want.

Comment (saad-ali) [inline hack/jenkins/e2e.sh]: Good point. Added `-f` to force the link, kept the `|| true` to make sure that we don't fail the build if something still manages to go wrong.

**Comment by saad-ali** [comment]
PTAL

**Comment by zmerlynn** [comment]
LGTM

**Pull request merged**